react-native/react/actions/login2.js

```diff
@@ -277,25 +163,302 @@ export function logout () {
+  return (dispatch, getState) => {
+    const provisionMethod = enums.provisionUi.ProvisionMethod.device
+    startLoginFlow(dispatch, getState, provisionMethod, null, null, Constants.actionRegisteredWithExistingDevice)
+  }
+}
+
+export function registerWithUserPass () {
+  return (dispatch, getState) => {
+    const title = 'Registering with your Keybase passphrase'
+    const subTitle = 'lorem ipsum lorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsum'
+    const provisionMethod = enums.provisionUi.ProvisionMethod.passphrase
+    startLoginFlow(dispatch, getState, provisionMethod, title, subTitle, Constants.actionRegisteredWithUserPass)
+  }
+}
+
+export function registerWithPaperKey () {
+  return (dispatch, getState) => {
+    const title = 'Registering with your paperkey requires your username'
+    const subTitle = 'Different lorem ipsum lorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsum'
+    const provisionMethod = enums.provisionUi.ProvisionMethod.paperKey
+    startLoginFlow(dispatch, getState, provisionMethod, title, subTitle, Constants.actionRegisteredWithPaperKey)
+  }
+}
+
+function startLoginFlow (dispatch, getState, provisionMethod, userPassTitle, userPassSubtitle, successType) {
+  const mobile = true // TODO desktop also
+  const deviceType = mobile ? 'mobile' : 'desktop'
+  const incomingMap = makeKex2IncomingMap(dispatch, getState, provisionMethod, userPassTitle, userPassSubtitle, successType)
+
+  engine.rpc('login.login', {deviceType}, incomingMap, (error, response) => {
+    dispatch({
+      type: successType,
+      error: !!error,
+      payload: error || null
+    })
+
+    dispatch(navigateTo([]))
+    dispatch(switchTab(DEVICES_TAB))
+  })
+}
+
+export function addANewDevice () {
+  return (dispatch, getState) => {
+    const provisionMethod = enums.provisionUi.ProvisionMethod.device
+    const userPassTitle = 'Registering a new device requires your Keybase username and passphrase'
+    const userPassSubtitle = 'reggy lorem ipsum lorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsumlorem ipsum'
+    const successType = null
+    const incomingMap = makeKex2IncomingMap(dispatch, getState, provisionMethod, userPassTitle, userPassSubtitle, successType)
+
+    engine.rpc('device.deviceAdd', {}, incomingMap, (error, response) => {
+      console.log(error)
+    })
+  }
+}
+
+function makeKex2IncomingMap (dispatch, getState, provisionMethod, userPassTitle, userPassSubtitle, successType) {
+  return {
+    'keybase.1.provisionUi.chooseProvisioningMethod': (param, response) => {
+      response.result(provisionMethod)
+    },
+    // TODO remove this when KEX2 supports not asking for this
+    'keybase.1.loginUi.getEmailOrUsername': (param, response) => {
+      const { username } = getState().login2.userPass
+      if (!username) {
+        dispatch(askForUserPass(userPassTitle, userPassSubtitle, () => {
```

Comment:
this works since userpass will store the user/pass in the store so the checks later (passphrase will be in there) will let it just response.result and not show the screen again